### PR TITLE
Demi test

### PR DESCRIPTION
**Pull request by demiculus**
A preview of the game is here: https://www.loom.com/share/c1fd70be12684777962b1a3385e74e36

- Red player uses `w` `s`
- Black player uses `i` `k`
- `r` resets the game
- `space` pauses the game
- Scoring can be seen from the console

- Did not do any of the extra TODOs

- Instead of making app.pause() I create a pause state inside gameState. Which is a variable. When calling gameState.pause = true game stops. When calling gameState.pause = false game continues.

- Sorry this is a long PR 25 commits. 

Future refactors that I didn't do:

- refactor the index.html so that it has no javascript code
  - create game.js and put the index.html js code there
  - put gameState.js in a variable in game.js
  - refactor collision functions to its own file

These would be enough for this scale, more reactors might be needed depending on what we're going to do, player count, ball count, ball speed, player skins/colors etc..

### CODE REVIEW

**Comment by Esqarrouth** [comment]
> Add keyboard functionalities for both players, W and S can be one player Up and Down can be for another player.

> Red player uses w s
> Black player uses i k

You haven't followed instructions

**Comment by Esqarrouth** [comment]
> r resets the game
> space pauses the game

> Add ball to the game, pressing SPACE button should start game, make sure ball bounces from boundries.

You haven't followed instructions
I would use p for pause

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules/
+package-lock.json
```
Comment (Esqarrouth) [inline .gitignore:2]: i thought these would be better in the project, but then it would be interesting to see how people deal with this

**Comment by Esqarrouth** [comment]
reading this, I get the idea that this guy knows his canvas

File: public/index.html [hunk 2]
```diff
@@ -30,6 +32,16 @@
 				height : plateHeight,
 				color  : 'black'
 			});
+
+			this.nodes.push({
+				id : 'ball',
```
Comment (Esqarrouth) [inline public/index.html]: I think a part of the task could be making ball circle?

File: public/index.html [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
 			plateSpeed = app.height/40
 			ballSpeed = plateSpeed / 2
-			ballXVelocity = 1
+			ballXVelocity = Math.random() > 0.5 ? -1 : 1
```
Comment (Esqarrouth) [inline public/index.html]: nice touch

File: public/index.html (hunk 2 above)
Comment (demiculus) [reply public/index.html]: Yes it'll make the person have to edit app.js & the rendering engine

**Comment by demiculus** [comment]
> You haven't followed instructions

I know, I need to lose points there.

I took the easy way. When I set the `up & down arrow`, the canvas was scrolling instead of the image in the canvas moving. Instead of figuring out how to do it, I just changed the keys.

@devcem just this behavior shows that I will take the easy way out and design the game according to the tech I can use, not develop the game according to the best design. 

This behavior is good in cases where it's an indie game. The developer doesn't have time to optimize for every detail just needs to get something out in front of the players. 

But this behavior is catastrophic when the company has millions of players and a team working on optimizing every single bit.

**Comment by devcem** [comment]
> > You haven't followed instructions
> 
> I know, I need to lose points there.
> 
> I took the easy way. When I set the `up & down arrow`, the canvas was scrolling instead of the image in the canvas moving. Instead of figuring out how to do it, I just changed the keys.
> 
> @devcem just this behavior shows that I will take the easy way out and design the game according to the tech I can use, not develop the game according to the best design.
> 
> This behavior is good in cases where it's an indie game. The developer doesn't have time to optimize for every detail just needs to get something out in front of the players.
> 
> But this behavior is catastrophic when the company has millions of players and a team working on optimizing every single bit.

Thank you for explanation. I improved repository with some changes. On review process I will make sure that all tasks clearly understood by candidates and review accordingly.

**Pull request closed without merging**